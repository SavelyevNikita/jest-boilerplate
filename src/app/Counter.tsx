
import * as React from 'react';

interface CounterProps {
    v: number;
}

interface CounterState {
    value: number;
}

export class Counter extends React.Component<CounterProps> {

    public state: CounterState = {
        value: this.props.v
    }

    private increaseCounter = (): void => {
        this.setState((state: CounterState): CounterState => {
            return {
                ...state,
                value: 1 + state.value
            };
        });
    }

    public render(): JSX.Element {
        return (
            <div>
                <span>{this.state.value}</span>
                <button onClick={this.increaseCounter}>Increase</button>
            </div>
        );
    }
}
