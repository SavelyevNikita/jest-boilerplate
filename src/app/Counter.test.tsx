
import * as React from 'react';
import { mount } from 'enzyme';
import { Counter } from './Counter';

test('Counter exists', (): void => {
    const el = (<Counter v={0} />);
    const sh = mount(el);
    expect(sh).toMatchSnapshot();
});

test('Counter has a button to increase the counter', (): void => {
    const el = (<Counter v={0} />);
    const sh = mount(el);
    expect(sh.find('button').text()).toBe('Increase');
});

test('Counter has an span to show the counter', (): void => {
    const el = (<Counter v={0} />);
    const sh = mount(el);
    expect(sh.find('span').text()).toBe('0');
});

test('If click on button then counter increases in one', (): void => {
    const el = (<Counter v={0} />);
    const sh = mount(el);
    sh.find('button').simulate('click');
    expect(sh.find('span').text()).toBe('1');
});

test('If click on button then counter increases in one begining from given value', (): void => {
    const el = (<Counter v={10} />);
    const sh = mount(el);
    sh.find('button').simulate('click');
    expect(sh.find('span').text()).toBe('11');
});
