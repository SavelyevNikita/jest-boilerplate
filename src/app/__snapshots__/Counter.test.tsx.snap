// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter exists 1`] = `
<Counter
  v={0}
>
  <div>
    <span>
      0
    </span>
    <button
      onClick={[Function]}
    >
      Increase
    </button>
  </div>
</Counter>
`;
